python/lsst/ts/externalscripts/maintel/tma/random_walk_and_take_image_gencam.py

```diff
@@ -0,0 +1,583 @@
+                  minimum: 0
+                  maximum: 2147483647
+              exp_times:
+                description: Exposure times used on each camera.
+                type: array
+                items:
+                  type: number
+                  minimum: 0
+              num_exp:
+                description: Number of exposures for all cameras.
+                type: integer
+                minimum: 1
+                default: 1
+              sleep_time:
+                description: Sleep time between exposures.
+                type: number
+                minimum: 0
+                default: 0
+              reason:
+                description: Optional reason for taking the data.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              program:
+                description: Optional name of the program this data belongs to, e.g. WFD, DD, etc.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              camera_playlist:
+                description: >-
+                  Optional name a camera playlist to load before running the script.
+                  This parameter is mostly designed to use for integration tests and is
+                  switched off by default (e.g. null).
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              max_dome_mount_az:
+                description: >-
+                  Maximum difference between dome position and target mount azimuth in degrees.
+                  If the actuall difference is larger than this parameter, the dome moves.
+                type: number
+                default: 10.
+              dome_offset:
+                description: Artificial offset for when the dome loses reference.
+                type: number
+                default: 0
+            required:
+            - camera_sal_indexes
+            - exp_times
+            - total_time
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def load_playlist(self):
+        """Load playlist."""
+        raise NotImplementedError()
+
+    async def move_dome(self, az: float):
+        """Move the dome to a give position
+
+        Parameters
+        ----------
+        az : float
+            New dome Azimuth position in degrees.
+        """
+        await self.tcs.rem.mtdome.cmd_exitFault.set_start()
+        await asyncio.sleep(5)
+        await self.tcs.rem.mtdome.cmd_moveAz.set_start(position=az, velocity=0)
+        await self.wait_for_dome_in_position()
+
+    async def random_walk_azel_by_time(self):
+        """Generate Az/El coordinates for a finite amount of time so we can
+        slew and track to these targets.
+        """
+        counter = 0
+        timer_task = asyncio.create_task(asyncio.sleep(self.config.total_time))
```

Comment:
I strongly suggest keeping the timer in the code that calls this. Then this can be a simple "get" function rather than a generator, which has several advantages:
* It eliminates the need to deal with a counter here -- all it does is return az/el -- which, in turn, lets you get rid of the counter field in the struct.
* By keeping time in the calling code, it becomes simple to abort the last iteration if the time runs out during it. I'm not saying you have to do that, but at least it becomes easy to do so if you decide in the future that you want to.

Also: while using a timer to keep track of when to quit is clever, I am uneasy about it for two reasons:
* It is likely to produce logged warnings if you abort the script, because this timer is still running. That  makes asyncio upset (though it doesn't break anything).
* It is harder to log how much time remains (at least in a way that is fully compatible with the timer) because timers don't make that information available.

A technique that avoids these pitfalls is to record the end time when you start running the script, then check the current time against it:
```
end_tai = utils.current_tai() + self.config.total_time
counter = 0
...
while utils.current_tai() < end_tai:
```
or, if you want to log remaining time (a bit uglier if you want to avoid computing it twice):
```
while True:
    rem_time = end_tai - utils.current_tai()
    if rem_time <= 0:
        return None
    self.log.info(f"Iteration {counter} begins; {rem_time:0.1f} seconds remaining")
```